Makefile

```diff
@@ -63,15 +71,25 @@ lib/libbase64.o: $(OBJS)
 	$(OBJCOPY) --keep-global-symbols=lib/exports.txt $@
 
 lib/config.h:
-	@echo "#define HAVE_AVX2   $(HAVE_AVX2)"    > $@
-	@echo "#define HAVE_NEON32 $(HAVE_NEON32)" >> $@
-	@echo "#define HAVE_NEON64 $(HAVE_NEON64)" >> $@
-	@echo "#define HAVE_SSSE3  $(HAVE_SSSE3)"  >> $@
-	@echo "#define HAVE_SSE41  $(HAVE_SSE41)"  >> $@
-	@echo "#define HAVE_SSE42  $(HAVE_SSE42)"  >> $@
-	@echo "#define HAVE_AVX    $(HAVE_AVX)"    >> $@
+	@echo "#define HAVE_AVX2                  $(HAVE_AVX2)"                   > $@
+	@echo "#define HAVE_NEON32                $(HAVE_NEON32)"                >> $@
+	@echo "#define HAVE_NEON64                $(HAVE_NEON64)"                >> $@
+	@echo "#define HAVE_SSSE3                 $(HAVE_SSSE3)"                 >> $@
+	@echo "#define HAVE_SSE41                 $(HAVE_SSE41)"                 >> $@
+	@echo "#define HAVE_SSE42                 $(HAVE_SSE42)"                 >> $@
+	@echo "#define HAVE_AVX                   $(HAVE_AVX)"                   >> $@
+	@echo "#define HAVE_FAST_UNALIGNED_ACCESS $(HAVE_FAST_UNALIGNED_ACCESS)" >> $@
+
+lib/table_generator: lib/table_generator.o
+	$(CC) $(CFLAGS) -o $@ $^
+
+lib/tables.h: lib/table_generator.c
+	@$(MAKE) -f $(THIS_FILE) lib/table_generator
+	./lib/table_generator > $@
+
```

Comment:
I've been rethinking a bit about this.
Writing
```Makefile
lib/tables.h: ./lib/table_generator
	$^ > $@
```
will make `lib/tables.h` to be rebuilt every time (well at least on fresh checkout or source tar.gz) since it depends on `lib/table_generator` which is not under source control and thus must be rebuilt.
This might be acceptable but won't allow for cross-compilation with an unpatched makefile since `lib/table_generator` won't be able to run on the build system. The option currently implemented in the PR would allow for cross-compilation out of the box.
The only time `lib/tables.h`and `lib/table_generator` would be rebuilt is if `lib/table_generator.c` changes and in this case, we can assume someone doing this knows what he is doing (i.e. knows cross-compilation won't work in that specific case).
I admit it's a bit more complex but allows for cross-compilation out of the box.
Your choice.